utils/check_docstrings.py

```diff
@@ -0,0 +1,1266 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility that checks all docstrings of public objects have an argument section matching their signature.
+
+Use from the root of the repo with:
+
+```bash
+python utils/check_docstrings.py
+```
+
+for a check that will error in case of inconsistencies (used by `make repo-consistency`).
+
+To auto-fix issues run:
+
+```bash
+python utils/check_docstrings.py --fix_and_overwrite
+```
+
+which is used by `make fix-copies` (note that this fills what it cans, you might have to manually fill information
+like argument descriptions).
+"""
+import argparse
+import ast
+import enum
+import inspect
+import operator as op
+import re
+from pathlib import Path
+from typing import Any, Optional, Tuple, Union
+
+from check_repo import ignore_undocumented
+
+from transformers.utils import direct_transformers_import
+
+
+PATH_TO_TRANSFORMERS = Path("src").resolve() / "transformers"
+
+# This is to make sure the transformers module imported is the one in the repo.
+transformers = direct_transformers_import(PATH_TO_TRANSFORMERS)
+
+OPTIONAL_KEYWORD = "*optional*"
+# Re pattern that catches args blocks in docstrings (with all variation around the name supported).
+_re_args = re.compile(r"^\s*(Args?|Arguments?|Attributes?|Params?|Parameters?):\s*$")
+# Re pattern that parses the start of an arg block: catches <name> (<description>) in those lines.
+_re_parse_arg = re.compile(r"^(\s*)(\S+)\s+\((.+)\)(?:\:|$)")
+# Re pattern that parses the end of a description of an arg (catches the default in *optional*, defaults to xxx).
+_re_parse_description = re.compile(r"\*optional\*, defaults to (.*)$")
+
+
+# This is a temporary list of objects to ignore while we progressively fix them. Do not add anything here, fix the
+# docstrings instead. If formatting should be ignored for the docstring, you can put a comment # no-format on the
+# line before the docstring.
+OBJECTS_TO_IGNORE = [
+    # Deprecated
+    "InputExample",
+    "InputFeatures",
+    # Signature is *args/**kwargs
+    # "PretrainedConfig", #ignored but could be fixed
+    # "GenerationConfig", #ignored but could be fixed
```

Comment:
```suggestion
```

Reply:
No I want to leave them here in the list of objects to try to fix.